homeassistant/components/lutron/__init__.py

```diff
@@ -128,6 +128,11 @@ def should_poll(self):
         """No polling needed."""
         return False
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"{self._controller.guid}_{self._lutron_device.uuid}"
```

Comment:
why do we need to combine it with the guid. Isn't the whole idea behind a uuid that it's unique?

Reply:
Also, how will this work with a device that has two entities represent different parts?

Reply:
The device UUID is just the unique id of the device in that config.  It's not a globally unique identifier across every lutron device in the world.  

The GUID of the controller uniquely identifies that controller globally.  So to produce a unique id for each entity, we need to append something to the GUID for the controller.

Reply:
I'm unsure of your second question.  There isn't an instance of a device that has two different entities in the current implementation.

Reply:
One only has to add a single battery powered device and we're going to have two entities :) 

Although a UUID might be only unique within the config, if it uses something like uuid4 spec, the odds of it being globally unique are 99.999%. 

What about `lutron_device.id`, how unique is that ?

Reply:
Can you clarify what you mean by adding a battery powered device that would have two entities?  I have pico remotes in my setup, and each button ends up with it's own entity with its own unique id.

lutron_device.id is a 3 or 4 digit number, that appears to be generated via some scheme based on the device type, and when it was added to the system.

Reply:
I think balloob is referring to the fact that in most integrations battery sensors have one or more sensor entities for their actual state (motion/temp/humidity/etc) plus a battery level sensor, or some switches have sensors for power draw or usage, etc. I'm not familiar with the `lutron` integration to know what all it supports though.

Reply:
Ahh I see.  Lutron doesn't have anything like that.  The closest thing are the keypads that end up effectively having multiple entities and they're modeled as scenes.

Reply:
Any other concerns with this PR?